### DISCUSSION TITLE
Methods Overview

### DISCUSSION THREAD

**Discussion by hanried** [earthlab-education/Earth-Analytics-AY24#392] [Q&A]
In the instructions for writing the portfolio post for our climate coding challenge, it says our analysis should include a researched methods overview. I'm not exactly sure what a methods overview would include. Do we just need to write out the steps we're doing? Can that be done through the comments we include at the beginning of our code cells or should it be written out more formally at the beginning of our code? 

I wasn't able to attend class on Thursday; apologies if this was covered in class & thanks in advance for any thoughts!

**Comment by BioNCphilic** [comment]
Thanks for the question. I'm interested to know the answer too.

**Comment by lauren-pease** [comment]
@hanried From what I understand, the methods section is used to indicate the libraries used and the library methods applied to transform the data, going step by step through your process.

**Comment by hanried** [comment]
Thanks @lauren-alexandra! That gives me a better idea of what we need to do. I think I'll go through my data cleaning/analysis steps and make sure each step has some language to explain my process.

**Comment by eculler** [comment]
Great questions, @hanried ! A methods overview could contain:

- A statement about what data processing steps you took and what model you're using if any.
- Some evaluation of the model assumptions. scikit-learn usually cites quite a few sources for each model they implement -- you could also look into those sources to bolster your choice of model if you're using one.
- Citations for Python (usually we cite the User Guide for the version you're using) and any 3rd party libraries you used (to the best of your ability -- we'll go over how to do that more later on, but just like data many libraries have a page on how to cite them.) You don't necessarily need an exhaustive list -- you could focus in on the scientific computing libraries such as geopandas and scikit-learn. It's likely more important to those authors to be cited in scientific publications.
- Any other studies you've found that use the same methods, with citations
